Makefile

```diff
@@ -150,6 +139,39 @@ clean:
 distclean: clean
 	rm -rf vendor/
 
+###############################################################################
+###                                Release                                  ###
+###############################################################################
+
+# create tag and run goreleaser without publishing
+create-release-dry-run:
+ifneq ($(strip $(TAG)),)
+	@echo "--> Dry running release for tag: $(TAG)"
+	@echo "--> Create tag: $(TAG) dry run"
+	git tag -s $(TAG) -m $(TAG)
+	git push --tags --dry-run
+	@echo "--> Delete local tag: $(TAG)"
+	@git tag -d $(TAG)
+	@echo "--> Running goreleaser"
+	TM_VERSION=$(TM_VERSION) goreleaser release --snapshot --clean
+	@rm -rf dist/
+	@echo "--> Done create-release-dry-run for tag: $(TAG)"
+else
+	@echo "--> No tag specified, skipping tag release"
+endif
+
+# create tag and publish it
+create-release:
+ifneq ($(strip $(TAG)),)
+	@echo "--> Running release for tag: $(TAG)"
+	@echo "--> Create release tag: $(TAG)"
+	git tag -s $(TAG) -m $(TAG)
+	git push --tags
```

Comment:
I think we should push a specific tag, last time doing release, i found i have some local tags of rc, which seem are deleted in remote ? if `git push --tags` all the local tags will be pushed.

Reply:
Good catch, thank you!

Changed to: `git push origin $(TAG)`